pkg/cloudprovider/provider/gce/cloudconfig.go

```diff
@@ -0,0 +1,93 @@
+//
+// Google Cloud Provider for the Machine Controller
+//
+
+package gce
+
+//-----
+// Imports
+//-----
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/kubermatic/machine-controller/pkg/ini"
+
+	"github.com/Masterminds/sprig"
+)
+
+//-----
+// Constants
+//-----
+
+// cloudConfigTemplate renders the cloud-config in gcfg format. All
+// fields are optional, that's why containing the ifs and the explicit newlines.
+// TODO(mue) Check for mandatory fields and better ways to drop empty fields,
```

Comment:
AFAIK there is no field which is mandatory

Reply:
Will reduce it down to the currently used ones. Also helps to drop these `if` statements in the template.